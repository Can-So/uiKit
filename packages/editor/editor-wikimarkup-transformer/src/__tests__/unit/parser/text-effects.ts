import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Text effects', () => {
  const testCases: Array<[string, string]> = [
    [
      'should find strong marks in the text',
      'This is a string with a *strong* text',
    ],
    [
      'should divide marks intervals into combination of marks',
      'This is a *string with a {color:red}bold red* text{color}.',
    ],
    [
      'should default unknown color to black #000000',
      'This is a unknown {color:nimabi}color{color}',
    ],
    ['should find monospace text', 'This is a string with {{monospaced text}}'],
    [
      'should not search for monospace text inside code macro',
      'This is a string with {code}mono {{space}} text{code}',
    ],
    ['should process superscript', 'This is a ^superscript^'],
    ['should process subscript', 'This is a ~subscript~'],
    ['should process emphasis', 'This is a _emphasis_'],
    ['should process citation', 'This is a ??citation??'],
    ['should process deleted', 'This is a -deleted-'],
    ['should process inserted', 'This is a +inserted+'],
    [
      'should process string with a wrong order of effects',
      'This is a *strong ^string* with a ~^bla*~~',
    ],
    [
      'should convert double baskslash to a hardBreak node',
      'this is a text with a\\\\new line in it',
    ],
    [
      'should not fail while applying marks to a string containing double backslash',
      'this is a text *with a\\\\new* line in it',
    ],
    [
      'should not apply marks on separate lines',
      `this is a line with an *asterisk
another one is here and it should* not be applied`,
    ],
    [
      'should render content inside code mark as plain text',
      'This is a {{monospace *bold* thingy}}',
    ],
    [
      "should use code mark first if it's the inner mark",
      'This is a *bold {{monospace}} thingy*',
    ],
    [
      'should not replace marks in the middle of the word',
      `https://app.datadoghq.com/screen/220506/product-fabric-content-service?tv_mode=true

https://app.datadoghq.com/screen/282018/product-fabric-adf-service?tv_mode=true#close`,
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
