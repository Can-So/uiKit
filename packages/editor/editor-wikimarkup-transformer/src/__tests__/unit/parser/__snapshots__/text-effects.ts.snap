// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Text effects should convert double baskslash to a hardBreak node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this is a text with a",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "new line in it",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should default unknown color to black #000000 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a unknown ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "color": "#000000",
              },
              "type": "textColor",
            },
          ],
          "text": "color",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should divide marks intervals into combination of marks 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a *string with a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "color": "#ff0000",
              },
              "type": "textColor",
            },
          ],
          "text": "bold red* text",
          "type": "text",
        },
        Object {
          "text": ".",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should find monospace text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a string with ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "monospaced text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should find strong marks in the text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a string with a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "strong",
          "type": "text",
        },
        Object {
          "text": " text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should not apply marks on separate lines 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this is a line with an *asterisk",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "another one is here and it should* not be applied",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should not fail while applying marks to a string containing double backslash 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this is a text ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "with a",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "new",
          "type": "text",
        },
        Object {
          "text": " line in it",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should not replace marks in the middle of the word 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "https://app.datadoghq.com/screen/220506/product-fabric-content-service?tv_mode=true",
              },
              "type": "link",
            },
          ],
          "text": "https://app.datadoghq.com/screen/220506/product-fabric-content-service?tv_mode=true",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "https://app.datadoghq.com/screen/282018/product-fabric-adf-service?tv_mode=true#close",
              },
              "type": "link",
            },
          ],
          "text": "https://app.datadoghq.com/screen/282018/product-fabric-adf-service?tv_mode=true#close",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should not search for monospace text inside code macro 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a string with ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "language": "java",
        "uniqueId": null,
      },
      "content": Array [
        Object {
          "text": "mono {{space}} text",
          "type": "text",
        },
      ],
      "type": "codeBlock",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process citation 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process deleted 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strike",
            },
          ],
          "text": "deleted",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process emphasis 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "emphasis",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process inserted 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "underline",
            },
          ],
          "text": "inserted",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process string with a wrong order of effects 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "strong ^string",
          "type": "text",
        },
        Object {
          "text": " with a ~^ bla*~~",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process subscript 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "type": "sub",
              },
              "type": "subsup",
            },
          ],
          "text": "subscript",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should process superscript 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "type": "sup",
              },
              "type": "subsup",
            },
          ],
          "text": "superscript",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should render content inside code mark as plain text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "monospace *bold* thingy",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Text effects should use code mark first if it's the inner mark 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "bold ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "monospace",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": " thingy",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
